sdk/core/azure-core-v2/src/main/java/com/azure/v2/core/credentials/TokenRequestContext.java

```diff
@@ -84,12 +88,13 @@ public TokenRequestContext addScopes(String... scopes) {
         Objects.requireNonNull(scopes, "The scopes parameter cannot be null.");
 
         if (scopes.length == 0) {
-            throw new IllegalArgumentException("At least one scope must be provided.");
+            throw LOGGER.logThrowableAsError(new IllegalArgumentException("At least one scope must be provided."));
         }
 
         for (String scope : scopes) {
-            if (scope == null) {
-                throw new IllegalArgumentException("Scopes cannot contain null values.");
+            if (CoreUtils.isNullOrEmpty(scope)) {
```

Comment:
why `CoreUtils` here and `ImplUtils` before? do we have too many copies of this helper?

Reply:
Core Utils in core v2 as public API, Impl Utils in client core as non public API.